setup/context/commands.go

```diff
@@ -408,29 +402,55 @@ func Endpoint(c *Context, syntax string, insecure, dev, quiet bool) error {
 }
 
 func Plan(c *Context, args []string) error {
-	if err := checkDockercfg(c, false); err != nil {
+	if err := checkDockercfg(c, c.InstanceDir, false); err != nil {
 		return err
 	}
 
 	args = append([]string{"plan"}, args...)
 	return Terraform(c, args)
 }
 
-func checkDockercfg(c *Context, force bool) error {
-	path := fmt.Sprintf("%s/dockercfg", c.InstanceDir)
+// checkDockercfg looks for docker config file (config.json or dockercfg) from
+// given directory and saves in `dockercfg` format in instance directory.
+func checkDockercfg(c *Context, dockerCfgDir string, force bool) error {
+	if dockerCfgDir == "" {
+		dockerCfgDir = c.InstanceDir
+	} else {
+		fmt.Printf("Checking for docker config file at: `%s`\n", dockerCfgDir)
+	}
+
+	// Load docker config file from given dockerCfgDir
+	configFile, err := LoadDockerConfig(dockerCfgDir)
```

Comment:
For the sake of keeping things simple, I don't think we need this "discovery" behavior. 
Specifying a file path via `--dockercfg=path/to/file` is all we need to implement.